src/main/java/fi/fmi/avi/archiver/spring/retry/RetryAdvice.java

```diff
@@ -0,0 +1,59 @@
+package fi.fmi.avi.archiver.spring.retry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.integration.handler.advice.RequestHandlerRetryAdvice;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.listener.RetryListenerSupport;
+import org.springframework.retry.support.RetryTemplateBuilder;
+
+import java.time.Duration;
+
+import static java.util.Objects.requireNonNull;
+
+public class RetryAdvice extends RequestHandlerRetryAdvice {
```

Comment:
Instead of subclassing for configuration, I think it might be structurally better to have a (static?) factory class returning configured vanilla `RequestHandlerRetryAdvice`, or simply a private factory method within `FileHandlerConfig`. What do you think?

As a side note, this code has similarities to `DataSourceConfig.databaseAccessRetryTemplate()`. But I'm not sure if it is (at least yet) worth deduplicating the code. Maybe then, if we somehow end up needing more retry configurations.

Reply:
👍